### PR TITLE
enhancement: More verbose summary

### PR DESCRIPTION
**Pull request by candiduslynx**
#### Summary

Addressing #951:
* Displaying errors, warnings and ignored issues info **only** if there were occurrences of such issues
* Add an extra footer suggesting running `cloudquery` with `--verbose` flag when ignored issues appear

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: pkg/ui/console/summary.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		ui.ColorizedOutput(ui.ColorHeader, summaryIgnored, ignored)
+		const footerIgnored = "\nProvider %s finished with %s ignored issues." +
+			"\nThis may be normal, however, you can use `--verbose` flag to see more details."
+		ui.ColorizedOutput(ui.ColorHeader, footerIgnored, key, ignored)
+	}
+
+	ui.ColorizedOutput(ui.ColorHeader, "\n\n")
+}
+
+func printIssues(diags diag.Diagnostics, severities ...diag.Severity) string {
```
Comment (disq) [inline pkg/ui/console/summary.go:60]: `formatDiagCounts` maybe? it doesn't seem to print anything

File: pkg/ui/console/summary.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	warnings := printIssues(diags, diag.WARNING)
+	if len(warnings) > 0 {
+		const summaryWarnings = "\t ⚠️ Warnings: %s"
+		ui.ColorizedOutput(ui.ColorHeader, summaryWarnings, warnings)
+	}
+
+	// ignored issues
+	ignored := printIssues(diags, diag.IGNORE)
+	if len(ignored) > 0 {
+		const summaryIgnored = "\t ❓ Ignored issues: %s"
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply pkg/ui/console/summary.go:50]: Why though?
As it's a really localised piece, defining `const` at the proper scope seems more logical.

[the remark this answers is not in the document]
Comment (erezrokah) [reply pkg/ui/console/summary.go:50]: That's a good point @candiduslynx. In this specific case should it be a const? It's used once in the scope so might be easier to follow the code if inline it, i.e. `ui.ColorizedOutput(ui.ColorHeader, "\t ❓ Ignored issues: %s", ignored)`

File: pkg/ui/console/summary.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	if summary.Name != summary.Alias {
+		key = summary.Name + `(` + summary.Alias + `)`
+	}
+	diags := summary.Diagnostics().Squash()
+	const (
+		fetchSummary = "Provider %s fetch summary: %s Total Resources fetched: %d"
+	)
+	ui.ColorizedOutput(
+		ui.ColorHeader,
+		fetchSummary,
```
Comment (erezrokah) [inline pkg/ui/console/summary.go:27]:
```suggestion
	ui.ColorizedOutput(
		ui.ColorHeader,
		"Provider %s fetch summary: %s Total Resources fetched: %d",
```

[non-blocking] Is this `const` used someplace else?

**Review by erezrokah — APPROVED** [review]
Thanks for the PR @candiduslynx, the code changes look good to me 🚀 

Added 2 non blocking comments.
Also, can you share a before/after screenshots? That would make it easier to understand the changes

**Review by roneli — APPROVED** [review]
LGTM, 1 nit maybe would add a little doc for each function

File: pkg/ui/console/summary.go (hunk 3 above)
Comment (candiduslynx) [reply pkg/ui/console/summary.go:27]: no, just this block of code

**Comment by candiduslynx** [comment]
> LGTM, 1 nit maybe would add a little doc for each function

added a small doc after rebase

**Comment by candiduslynx** [comment]
> Also, can you share a before/after screenshots? That would make it easier to understand the changes

* before:
  image
  Notice that errors part is printed out even though there are no errors during fetch.
* after:
  image
  The issues will be printed out only if there are some (and only for the appropriate levels).

**Review by erezrokah — APPROVED** [review]

**Review by disq — APPROVED** [review]

**Pull request merged**